Implement landing page design from Figma

PrachodK opened the pull request:
Implemented new landing page design from Figma with smooth section transitions.

PrachodK committed:
Implement landing page design from Figma

miguelaenlle reviewed (changes requested):
Overall looks good @PrachodK ! Here are some suggestions:

- [ ] 1. Export the background as an SVG. It looks kind of blurry on my end. 

- [ ] 2. Most of the background isn't visible on mobile: 

<img width="407" height="626" alt="Image" src="https://github.com/user-attachments/assets/7a273e09-4350-4b13-ac57-4e31e8b98311" />

- [x] 3. Long-form content is too small on mobile: 

<img width="410" height="688" alt="Image" src="https://github.com/user-attachments/assets/09fc6b3c-85f3-4f1f-8c16-557af76d85fd" />

<img width="400" height="464" alt="Image" src="https://github.com/user-attachments/assets/00c8ad0c-506e-41d4-b9e0-07546af9c3bd" />

I recommend doubling the font size while adhering to the designs. E.g. TreeHacks uses `font-size: 16px`:

<img width="404" height="743" alt="Image" src="https://github.com/user-attachments/assets/70c12253-86fc-47f8-8f8b-520ff74e4d9b" />

Nits:

- [x] 4. Reduce the Hackstronaut font size, it takes up a considerable portion of the screen. I think the design takes up a better percentage. 

Current:
<img width="1455" height="815" alt="Screenshot 2025-11-18 at 9 04 09 AM" src="https://github.com/user-attachments/assets/43ebf71f-1cea-45e4-a60e-3d2ed74be2f2" />

On Figma:
<img width="764" height="353" alt="Image" src="https://github.com/user-attachments/assets/ed973c34-7edb-498d-9226-58efd33ff744" />


- [x] 5. Similarly I recommend following the gradient button that design created. I also recommend using Material UI for this, like we're doing on registration.

<img width="717" height="248" alt="Image" src="https://github.com/user-attachments/assets/6ebcf9e4-5329-4ac6-9bde-787377420bd7" />

- [x] 6. Nit: Left-align the "about event" content, will be helpful when we put an image where the gray box is. Also, reduce the vertical padding on the section 

<img width="1005" height="590" alt="Image" src="https://github.com/user-attachments/assets/827754c2-69d3-4723-ae84-0c95c56669e5" />

- [x] 7. Nit: Align section headers with their corresponding content

<img width="681" height="539" alt="Image" src="https://github.com/user-attachments/assets/bea6b4c6-f725-4339-8ab3-c0b802c08696" />

<img width="408" height="342" alt="Image" src="https://github.com/user-attachments/assets/e55793bc-cd25-4ced-a49e-3d38d96e16d2" />

Extra changes (we can do these later)
- Animate a drop-in for the Hackstronaut title 
- Refactor the SCSS code in MUI (let's wait for the final designs to come first)

PrachodK committed:
Polish landing page and add challenge page

Improved landing page mobile responsiveness with better font sizes, updated Register button to use Material UI with gradient design, added drop-in animation for hero title, and enhanced section layouts. Also set up the design challenge page at /register/challenge with public access.

PrachodK committed:
Update landing page background and ellipse positioning

miguelaenlle commented:
Moved changes to https://github.com/HackIllinois/site/pull/344
